### PR TITLE
Billy/scavenge

### PR DESCRIPTION
**Pull request by okwme**
Added new scavenger hunt tutorial.

### CODE REVIEW

**Comment by okwme** [comment]
I'm giving a workshop next Wednesday called `Digital Scarcity & Gaming in Golang` and thought that I should just turn it into a tutorial since that's basically what it is. I'm not sure what is needed to integrate into the rest of the tutorial publication flow, but I know it needs some revision i have only kind of run through it myself.

I know it's last minute but if anyone was able to run through it and give me feedback by EOD Monday I could incorporate it on Tuesday and have it ready to present on Wednesday 🌠

File: scavenge/tutorial/01-intro.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Digital Scarcity and Gaming in Golang
+## Workshop by Billy Rennekamp
+
+### Hello and welcome to the text version of this workshop!
```
Comment (marbar3778) [inline scavenge/tutorial/01-intro.md]: we should use this wording for onboarding new comers to blockchain!!

File: scavenge/x/scavenge/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# TODO
+
```
Comment (marbar3778) [inline scavenge/x/scavenge/README.md]: can be removed

File: scavenge/x/scavenge/abci.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package scavenge
```
Comment (marbar3778) [inline scavenge/x/scavenge/abci.go]: this can be removed

File: scavenge/x/scavenge/alias.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package scavenge
+
+import (
+	"github.com/okwme/scavenge/x/scavenge/internal/keeper"
+	"github.com/okwme/scavenge/x/scavenge/internal/types"
+)
+
+const (
+	// TODO: define constants that you would like exposed from the internal package
```
Comment (marbar3778) [inline scavenge/x/scavenge/alias.go]: can remove all todos

File: scavenge/x/scavenge/internal/types/params.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
+package types
```
Comment (marbar3778) [inline scavenge/x/scavenge/internal/types/params.go]: can be removed

File: scavenge/x/scavenge/abci.go (hunk 3 above)
Comment (okwme) [reply scavenge/x/scavenge/abci.go]: Think the definitions of `BeginBlocker` and `EndBlocker` are needed even tho they aren't doing anything

File: scavenge/x/scavenge/README.md (hunk 2 above)
Comment (okwme) [reply scavenge/x/scavenge/README.md]: :+1:

File: scavenge/x/scavenge/internal/keeper/params.go [hunk 6]
```diff
@@ -0,0 +1 @@
+package keeper
```
[the remark this answers is not in the document]
Comment (okwme) [reply scavenge/x/scavenge/internal/keeper/params.go]: :+1:

File: scavenge/x/scavenge/alias.go (hunk 4 above)
Comment (okwme) [reply scavenge/x/scavenge/alias.go]: :+1:

File: scavenge/x/scavenge/internal/types/params.go (hunk 5 above)
Comment (okwme) [reply scavenge/x/scavenge/internal/types/params.go]: think i need some of these functions even tho they dumb dumb

**Comment by okwme** [comment]
> do you want to add comments at the top of the tutorial so it can work with the vuepress theme

Yes! I see the other tutorials have this:
```
---
order: 5
---
```
at the top but i don't know what it means or what it does.

**Comment by okwme** [comment]
ok added those headers just using the `order: x` as page number

also added the snippet snytax for vuepress but left the code in commented out html just as fallback

File: scavenge/tutorial/04-messages.md [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+## MsgCreateScavenge
+Messages are `types` which live inside the `./x/scavenge/internal/types/` directory. There is already a `msg.go` file but we will make a new file for each Message type. We can use `msg.go` as a starting point by renaming it to `MsgCreateScavenge.go` like: 
+```bash
+# Assuming your current working directory is the root of your application
+mv ./x/scavenge/internal/types/msg.go  ./x/scavenge/internal/types/MsgCreateScavenge.go
+```
+Inside this new file we will uncomment and follow the instructions of renaming variables until it looks as follows:
+```go
+package types
```
[the remark this answers is not in the document]
Comment (okwme) [reply scavenge/tutorial/04-messages.md]: :+1:

**Comment by okwme** [comment]
Do I need to add it somewhere else to be included in the Tutorials sidebar?

**Comment by okwme** [comment]
I see there's a deploy-website CI command and found this address within the logs:
```
https://cloudfront.amazonaws.com/2019-03-26/distribution/*************/invalidation/I2J5JQHU3NKVZF
```
but without the token key i don't think I can see it

is there any way to preview it? maybe @fadeev  knows?

**Comment by fadeev** [comment]
@okwme I've merged master into this branch and added to `config.js`:

```
          {
            title: 'Scavenge',
            path: '/scavenge/tutorial/',
            directory: true
          }
```

So it will show up in the sidebar.

**Comment by fadeev** [comment]
To see how it looks, if you have `npm` installed, you can run `npm install && npm run dev` (if not, ping me and I'll upload it to now.sh). We don't have staging for this repo and netlify isn't connected either.

**Comment by okwme** [comment]
thanks @fadeev !
I ran it locally and made some tweaks.
I think it all looks good now : )
Anything else needed before it's ok to merge?

**Comment by fadeev** [comment]
Looks good to me 💯

**Comment by okwme** [comment]
also: does merging build and deploy the site automatically or does that need to happen somewhere else?

**Pull request merged**